File: JavaCode/src/MainViewController.java
/**
 * Created by akkey on 2017/06/11.
 */
public class MainViewController extends BaseViewController {
    UIButton button1;
    UIButton button2;
    UIButton button3;
    UIImageView backImageView;

    @Override
    void viewDidLoad() {
        setBKImage(backImageView);

        button1.customSetting(this, "Button1");
        button1.tag = 1;
        button2.customSetting(this, "Button2");
        button2.tag = 2;
        button3.customSetting(this, "Button3");
        button3.tag = 3;
    }

    @Override
    void onClickButton(UIButton sender) {
        final String name;

        switch(sender.tag) {
            case 1:
                name = "01";
                break;
            case 2:
                name = "02";
                break;
            case 3:
                name = "03";
                break;
            default:
                name = "04";
        }
        // abbreviate
    }
}

File: PhotoGurashi/ViewController/MainViewController.swift
//
//  MainViewController.swift
//  PhotoGurashi
//
//  Created by AKIO on 2017/04/23.
//  Copyright © 2017 AkkeyLab. All rights reserved.
//

import UIKit

class MainViewController: BaseViewController {
    @IBOutlet weak var button1: UIButton!
    @IBOutlet weak var button2: UIButton!
    @IBOutlet weak var button3: UIButton!
    @IBOutlet weak var backImageView: UIImageView!
    
    override func viewDidLoad() {
        setBKImage(backImageView)

        button1.customSetting(self, title: "Button1")
        button1.tag = 1
        button2.customSetting(self, title: "Button2")
        button2.tag = 2
        button3.customSetting(self, title: "Button3")
        button3.tag = 3
    }

    override func onClickButton(_ sender: UIButton) {
        let name: String

        switch sender.tag {
        case 1:
            name = "01"
        case 2:
            name = "02"
        case 3:
            name = "03"
        default:
            name = "04"
        }
        let vc = CameraViewController()
        vc.view.backgroundColor = UIColor.white
        vc.setCharaImage(name)
        self.present(vc, animated: true, completion: nil)
    }
}
